src/ui/SolidityTypeFormatter.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+var yo = require('yo-yo')
+
+module.exports = {
+  formatData: formatData,
+  extractData: extractData
+}
+
+function formatData (key, data) {
+  var style = fontColor(data)
+  var type = ''
+  if (data.type !== 'array' && data.type !== 'struct') {
+    type = data.type
+  }
+  return yo`<label>${key}: <label style=${style}>${data.self}</label><label style='font-style:italic'> ${type}</label></label>`
+}
+
+function extractData (item, key) {
+  var ret = {}
+  if (item.type === 'array') {
+    ret.children = item.value || []
+    ret.self = 'Array' + '[' + ret.children.length + ']'
```

Comment:
Can we make it more specific, i.e. `struct MyStruct` or `uint[12][]`?
Didn't we want to put the length as a "pseudo-child" into the list of children?